src/malli/instrument.clj

```diff
@@ -23,7 +23,7 @@
                                           (true? (:gen d)) (dissoc $ :gen)
                                           :else $))]
                          (alter-meta! v assoc ::original-fn original-fn)
-                         (alter-var-root v (constantly (m/-instrument dgen original-fn)))
+                         (alter-var-root v (constantly (m/-instrument dgen original-fn nil :original-fn-meta (meta v))))
```

Comment:
we discussed this briefly with @ikitommi and we'd prefer if you passed in `:meta (meta v)` as part of `dgen`, the first argument of `-instrument`